### DISCUSSION TITLE
Create row in database using a dataclass

### DISCUSSION THREAD

**Discussion by peterhijma** [simolus3/drift#1316] [Q&A]
I was always using this to create (or update) a database row:

```dart
/// Instantiation of a dataclass generated by moor
Foo(
  id: id,  // Possibly null, if it's a new instance
  bar: "baz",
);

/// Method of DAO:
Future<void> insertOrUpdateFoo(Foo foo) {
  return into(foos).insertOnConflictUpdate(foo);
}
```

But this can't be used anymore (since null safety), because `id` is an `int id` and therefore not nullable.

I'm kinda stuck. What's the best way to fix this?

**Comment by simolus3** [comment]
The best way to fix this is to use [companions](https://moor.simonbinder.eu/docs/getting-started/writing_queries/#updates-and-deletes) generated by moor for specifically this purpose. Moor will generate a `FooCompanion.insert` where fields required for an insert are marked as `required`.

In places where you have been using `Foo` now but want to support both `Foo` and `FooCompanion` (your DAO method is a good example of that), use `Insertable<Foo>` instead.

**Reply by peterhijma** [reply]
Thank you @simolus3 I think I've got it!
